x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/parser/ExpressionTests.java

```diff
@@ -64,4 +72,14 @@ public void testLiteralIntegerWithByteValue() throws Exception {
         assertEquals(Integer.valueOf(Byte.MAX_VALUE), l.value());
         assertEquals(DataType.INTEGER, l.dataType());
     }
+
+    public void testLiteralIntegerInvalid() throws Exception {
+        ParsingException ex = expectThrows(ParsingException.class, () -> parser.createExpression("123456789098765432101"));
+        assertEquals(ex.getErrorMessage(), "Cannot parse number [123456789098765432101]");
```

Comment:
Shouldn't this be a "Number X is too large" situation?

Reply:
Fixed that - makes the parsing code a bit more ugly but the description is more accurate.